### ISSUE TITLE
Cannot use aliases inside hooks

### ISSUE THREAD

**Issue by cdwmhcc** [nuxt/nuxt#25358]
### Environment

------------------------------
- Operating System: Windows_NT
- Node Version:     v20.9.0
- Nuxt Version:     3.9.3
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  pnpm@8.14.0
- Builder:          -
- User Config:      runtimeConfig, alias, css, imports, components, modules, i18n, hooks, devtools
- Runtime Modules:  @nuxtjs/i18n@8.0.0
- Build Modules:    -
------------------------------

### Reproduction

Create file config.ts:

``` ts
export const config = {}
```

Create file hooks/build-before.ts:

``` ts
import { config } from '~/config'

export const buildBefore = async() => {
     console.log(config)
}
```

In nuxt.config.ts:

``` ts
import { buildBefore } from './hooks/build-before'
export default defineNuxtConfig({
   hooks: {
     'build:before': async () => {
       await buildBefore()
     },
   },
})
```

This will report an error, indicating that the config module cannot be found. 

Change import { config } from '~/config' in `'~config'` to `'../config'` and it will work fine

### Describe the bug

In the files associated with functions imported in nuxt.config.ts hooks, relative paths must be used instead of aliases.

**Comment by markthree** [comment]
Because the internal loadNuxtConfig does not currently support aliases，But I think we should be able to support it in the future

**Comment by cdwmhcc** [comment]
> Because the internal `loadNuxtConfig` does not currently support aliases，But I think we should be able to support it in the future

Thank you for your quick response.

**Comment by manniL** [comment]
@markthree This is a bit tricky as *custom* aliases could be defined in the `nuxt.config.ts`. So we have a "chicken/egg" problem here IMO

**Comment by cdwmhcc** [comment]
@markthree @manniL 

There is a derivative problem here.

Create file /hooks/build-before/index.ts :

``` ts
export const buildBefore = async () => {
    console.log('test')
}
```

In the nuxt.config.ts file:
``` ts
import { buildBefore } from './hooks/build-before'
export default defineNuxtConfig({
   hooks: {
     'builder:watch': async (_, path) => {
       if (path.startsWith('hooks/build-before/')) {
         await buildBefore()
       }
     },
   },
})
```

When I modify `console.log('test')` in /hooks/build-before/index.ts to `console.log('test1')`, it does re-execute the `build-before` function, but it executes caching Instead of the latest build-before function.

**Comment by markthree** [comment]
@manniL 
Thanks, I found it too.

If one wants to solve this problem
1. load nuxt.config.ts and stubbornly use your own set of alias presets (easiest way); 
2. we can give higher priority to config files (chicken/egg ancestry).

Regarding point 2, I think it ends up in the same problematic dead end loop (ancestor of ancestor of ancestor ......) .

So at the end of the day, this seems to be a philosophical issue, not a technical one. 🤣

**Comment by markthree** [comment]
@cdwmhcc 
Regarding other file changes not triggering restart, I think this is a bug, could you help to raise one more issue?

This is a simple solution (you can use it like this for now) 👇

```ts
import { buildBefore } from "./hooks/build-before";

export default defineNuxtConfig({
  hooks: {
    "build:before"() {
      buildBefore();
    },
  },
  watch: [/hooks\/.*/], // watch Required directories (Trigger restart on change)
});
```

**Comment by cdwmhcc** [comment]
> file changes not triggering restart

Changes to files under hooks/build-before can be monitored normally.
But it cannot apply the latest file changes, it triggers the cache.

For example, in hooks/build-before/index.ts, I pass console.log('test') and modify it to console.log('test1'). It still prints test, not the latest test1.

**Comment by cdwmhcc** [comment]
If you follow your solution and use `watch: [/hooks\/.*/]` directly, it will restart nuxt. Each restart will take too long, and the development efficiency will be very low.

Currently I have installed `nodemon` and use it to monitor. But this is a third-party integration after all, and there are some inconveniences.

My initial understanding of `build:watch` is that it implements the same function as nodemon, haha.

**Comment by markthree** [comment]
@cdwmhcc 
Actually, updating `./hooks/build-before` doesn't trigger a restart, whereas here the trigger is always `buildBefore` (old)

**Comment by markthree** [comment]
> If you follow your solution and use `watch: [/hooks\/.*/]` directly, it will restart nuxt. Each restart will take too long, and the development efficiency will be very low.
> 
> Currently I have installed `nodemon` and use it to monitor. But this is a third-party integration after all, and there are some inconveniences.
> 
> My initial understanding of `build:watch` is that it implements the same function as nodemon, haha.

Yes, it does lead to inefficient development if changes are made frequently.

If you want hot updates to take care of that, maybe you can use [jiti](https://github.com/unjs/jiti)

**Comment by markthree** [comment]
> If you want hot updates to take care of that, maybe you can use [jiti](https://github.com/unjs/jiti)
```ts
import jiti from "jiti";

function load(path: string) {
  // @ts-ignore
  return jiti(null, {
    esmResolve: true,
    interopDefault: true,
    requireCache: false,
  })(
    path,
  );
}

export default defineNuxtConfig({
  hooks: {
    async "builder:watch"() {
      const { buildBefore } = await load("./hooks/build-before");
      buildBefore();
    },
  },
});
```

**Comment by danielroe** [comment]
This now works via https://github.com/nuxt/nuxt/pull/28008!
